libsolidity/interface/CompilerStack.cpp

```diff
@@ -1206,13 +1206,93 @@ Json CompilerStack::ethdebug(Contract const& _contract, bool _runtime) const
 	solAssert(m_stackState >= AnalysisSuccessful, "Analysis was not successful.");
 	solAssert(_contract.contract);
 	solUnimplementedAssert(!isExperimentalSolidity());
-	if (_runtime)
+	Json result = {
+		{
+			"contract",
+			{
+					{"name", _contract.contract->name()},
+					{
+						"definition",
+						{{
+							"sources",
```

Comment:
Assuming this is from [Source range schema](https://ethdebug.github.io/format/spec/materials/source-range/), shouldn't this be singular?

```suggestion
							"source",
```

Reply:
Also, what's up with the double braces? Doesn't this create a one-element array or something?

You also have a weird double indent under `"contract"`.

Reply:
with nlohmann json double braces are needed to ensure that an object is created - if not it will be an array

Reply:
Hmm... is this why you now replaced it all with assignments and manually created objects? :)